rust/ql/lib/codeql/rust/dataflow/internal/DataFlowImpl.qll

```diff
@@ -0,0 +1,324 @@
+/**
+ * Provides Rust-specific definitions for use in the data flow library.
+ */
+
+private import codeql.util.Void
+private import codeql.dataflow.DataFlow
+private import codeql.dataflow.internal.DataFlowImpl
+private import rust
+private import codeql.rust.controlflow.ControlFlowGraph
+private import codeql.rust.dataflow.Ssa
+
+module Node {
+  /**
+   * An element, viewed as a node in a data flow graph. Either an expression
+   * (`ExprNode`) or a parameter (`ParameterNode`).
+   */
+  abstract class Node extends TNode {
+    /** Gets the location of this node. */
+    abstract Location getLocation();
+
+    /** Gets a textual representation of this node. */
+    abstract string toString();
+
+    /**
+     * Gets the expression that corresponds to this node, if any.
+     */
+    Expr asExpr() { none() }
```

Comment:
The return type here should be `CfgNode`, because we will base data flow on the CFG and not on the AST.

Reply:
Let's leave it at `Expr`, for now, and then change it once we have a proper `Expr` layer on top of `CfgNode`.